.github/bin/download-maven-dependencies.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Enable all known profiles to make sure their plugin dependencies are downloaded
+# GIB should be disabled even though it's profile is activated, to make sure it doesn't skip any submodules
+$RETRY $MAVEN -P ci,errorprone-compiler ${MAVEN_GIB} -Dgib.disable dependency:go-offline
+$RETRY $MAVEN -P ci,errorprone-compiler ${MAVEN_GIB} -Dgib.disable de.qaware.maven:go-offline-maven-plugin:resolve-dependencies
+# trino-pinot overrides some common dependency versions, focus on it to make sure those overrides are downloaded as well
+$RETRY $MAVEN -P ci,errorprone-compiler ${MAVEN_GIB} -Dgib.disable de.qaware.maven:go-offline-maven-plugin:resolve-dependencies -pl ':trino-pinot'
```

Comment:
I am concerned about this line. Why do we need it?
Are both `dependency:go-offline` and `de.qaware.maven:go-offline-maven-plugin:resolve-dependencies` unable to download two different versions of a library, if two versions are needed for the build?
That would be a major 🤦‍♂️ if this was true. And it wouldn't be sufficient to just single out pinot, since any module can declare anything it needs, right?

Reply:
> I am concerned about this line. Why do we need it? Are both `dependency:go-offline` and `de.qaware.maven:go-offline-maven-plugin:resolve-dependencies` unable to download two different versions of a library, if two versions are needed for the build? That would be a major 🤦‍♂️ if this was true. And it wouldn't be sufficient to just single out pinot, since any module can declare anything it needs, right?

It seems that https://github.com/qaware/go-offline-maven-plugin/issues/23 affects all dependencies, but the fix introduced in `1.2.8` only changed `Set<Plugin>` into `List<Plugin>`. We could submit a PR, but using both tools fixes this faster.

Reply:
if we consider that project maintained, yes, we should report a bug and perhaps submit a PR.
if  we consider that project unmaintained, we shouldn't use it at all. I would assume it will break sooner or later eg due to maven or jdk changes.

Reply:
Downloading dependencies is an additive process, so I would argue any tool helps, even if it's not perfect.
We want to have as many dependencies cached as possible, but we don't need 100% dependency coverage.
As long as the number of downloads during build is going down, chances of a Maven Central failure taking down the build also decrease.

Reply:
> Downloading dependencies is an additive process, so I would argue any tool helps, even if it's not perfect.

we don't need perfect tools, but we need maintained ones.
an unmaintained tool may likely cause head-aches in the future

anyone, we can fix that problem later, when it shows up

> We want to have as many dependencies cached as possible, but we don't need 100% dependency coverage.

I feel like I am not fully convinced yet. 
I think we really want 100% coverage, otherwise we will be flaky for ever.

Reply:
I think we've run the experiment to the end, and the current version is capable of running in `--offline` mode.
I'm not sure how this will impact maintainability.
If a new dependency isn't detected by any of the our tools, then we might have to fall back on adding more `<dynamicDependencies>`